packages/react/ds-app-launchpad/src/ui/FileTree/Provider.tsx

```
@@ -0,0 +1,96 @@
+import type React from "react";
+import { useCallback, useReducer } from "react";
+import Context from "./Context.js";
+import "./styles.css";
+import type {
+  FileSelectionOptions,
+  ProviderOptions,
+  SearchOptions,
+} from "./types.js";
+
+type State = {
+  internalSearchQuery: SearchOptions["searchQuery"];
+  internalSelectedFile: FileSelectionOptions["selectedFile"];
+};
+
+type Action =
+  | { type: "SEARCH"; payload: SearchOptions["searchQuery"] }
+  | { type: "SELECT_FILE"; payload: FileSelectionOptions["selectedFile"] };
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "SEARCH":
+      return { ...state, internalSearchQuery: action.payload };
+    case "SELECT_FILE":
+      return { ...state, internalSelectedFile: action.payload };
+    default:
+      return state;
```

Should this throw an error if an unexpected action type?